apps/builder/app/builder/features/sidebar-left/panels/components/components.tsx

```diff
@@ -163,43 +58,52 @@ export const TabContent = ({ publish, onSetActiveTab }: TabContentProps) => {
         title="Add"
         suffix={<CloseButton onClick={() => onSetActiveTab("none")} />}
       />
-      <Flex
-        gap="1"
-        wrap="wrap"
-        css={{ padding: theme.spacing[3], overflow: "auto" }}
-        ref={useDragHandlers.rootRef}
-      >
-        {listedComponentNames.map((component: Instance["component"]) => {
-          const meta = getComponentMeta(component);
-          if (meta === undefined) {
-            return null;
-          }
-          return (
-            <ComponentCard
-              onClick={() => {
-                onSetActiveTab("none");
-                const selectedPage = selectedPageStore.get();
-                if (selectedPage === undefined) {
-                  return;
-                }
-                const dropTarget = findClosestDroppableTarget(
-                  instancesStore.get(),
-                  // fallback to root as drop target
-                  selectedInstanceSelectorStore.get() ?? [
-                    selectedPage.rootInstanceId,
-                  ]
-                );
-                insertNewComponentInstance(component, dropTarget);
-              }}
-              data-drag-component={component}
-              label={meta.label}
-              icon={<meta.Icon />}
-              key={component}
+      <div ref={draggableContainerRef}>
+        {Array.from(componentCategories).map((category) => (
+          <CollapsibleSection label={category} key={category} fullWidth>
+            <ArrowFocus
+              render={({ handleKeyDown }) => (
+                <Flex
+                  onKeyDown={handleKeyDown}
+                  gap="2"
+                  wrap="wrap"
+                  css={{ px: theme.spacing[9], overflow: "auto" }}
+                >
+                  {(metasByCategory.get(category) ?? []).map(
+                    (meta: WsComponentMeta, index) => {
+                      const component = componentNamesByMeta.get(meta);
+                      if (component === undefined) {
+                        return null;
+                      }
+                      return (
+                        <Tooltip content={meta.label} key={component}>
+                          <ComponentCard
+                            onClick={() => {
+                              onSetActiveTab("none");
+                              handleInsert(component);
+                            }}
+                            onKeyDown={(event) => {
```

Comment:
Doesn't enter trigger click? Kinda onPress stuff. Maybe try `@react-aria/press`?

Reply:
it doesn't trigger click, probably that behavior exists only for buttons? haven't really looked into that package `@react-aria/press` maybe we should start using it all the time?

Reply:
done